websites/W/Worldhideaway/presence.ts

```diff
@@ -2,151 +2,177 @@ const presence = new Presence({
 		clientId: "815515385326469131",
 	}),
 	browsingTimestamp = Math.floor(Date.now() / 1000);
-let search: HTMLInputElement, title: HTMLElement;
 
 presence.on("UpdateData", async () => {
 	const presenceData: PresenceData = {
-			largeImageKey: "logo",
+			largeImageKey: "https://i.imgur.com/P3S7UhZ.png",
 			startTimestamp: browsingTimestamp,
 		},
-		page = window.location.pathname,
-		expage = window.location.href;
-	switch (page) {
-		case "/": {
-			search = document.querySelector(
-				"#masthead > div.header-search-wrap > div > form > label > input"
-			);
-			if (!search || search.value === "")
-				presenceData.details = "Viendo Inicio";
-			else {
-				presenceData.details = "Buscando:";
-				presenceData.state = search.value;
-				presenceData.smallImageKey = "searching";
-			}
-
-			break;
-		}
-		case "/activity/": {
-			presenceData.details = "Leyendo Sobre:";
-			presenceData.state = "Actividad";
-			presenceData.smallImageKey = "reading";
+		search = document.querySelector<HTMLInputElement>(
+			"#masthead > div.header-search-wrap > div > form > label > input"
+		),
+		{ href, pathname } = document.location,
+		[privacy, buttons, covers] = await Promise.all([
+			presence.getSetting<boolean>("privacy"),
+			presence.getSetting<boolean>("buttons"),
+			presence.getSetting<boolean>("covers"),
+		]);
+	if (privacy) presenceData.details = "Hojeada";
+	else {
+		switch (pathname.split("/")[1]) {
+			case "": {
+				if (search?.value) {
+					presenceData.details = "Buscando";
+					presenceData.state = search.value;
+					presenceData.smallImageKey = "searching";
+				} else presenceData.details = "Viendo Inicio";
 
-			break;
-		}
-		case "/members/": {
-			presenceData.details = "Viendo:";
-			presenceData.state = "Miembros";
-
-			break;
-		}
-		case "/forums/": {
-			presenceData.details = "Viendo:";
-			presenceData.state = "Foro";
+				break;
+			}
+			case "register": {
+				presenceData.details = "Registro";
+				break;
+			}
+			case "logros": {
+				presenceData.details = "Viendo logros";
+				presenceData.buttons = [
+					{
+						label: "Viendo Logros",
+						url: href,
+					},
+				];
+				break;
+			}
+			case "events": {
+				presenceData.details = "Viendo events";
+				presenceData.buttons = [
+					{
+						label: "Viendo Events",
```

Comment:
The label is the text that shows up on the button. Its purpose is to perform an action so you should be using the infinitive form instead of the gerund (Ver Eventos)

Reply:
Thanks for everything. I asked one of my buddies to translate but I think he messed up 😂